webboot/wget_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestWget(t *testing.T) {
+	type test struct {
+		name    string
+		isolink string
+		md5link string
+	}
+	tests := []test{
+		{name: "tinyCoreISO", isolink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", md5link: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso.md5.txt"},
+		{name: "archLinuxISO", isolink: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", md5link: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/md5sums.txt"},
+		{name: "ubuntuISO", isolink: "http://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso", md5link: "http://releases.ubuntu.com/bionic/MD5SUMS"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			r, err := wget(v.isolink)
+			if err != nil {
+				t.Errorf("Failed to receive the ISO: got %v, want nil", err)
+			}
+
+			body, err := ioutil.ReadAll(r)
+			if err != nil {
+				t.Errorf("Failed to read the ISO: got %v, want nil", err)
```

Comment:
why would I put a continue after this? t.Errorf recognizes that this is an error and still executes the rest of the code. Also, when I attempt to write a "continue" after t.Errorf, it gives me an error saying "continue is not in a loop"

Reply:
Instead, the error message should include t.Fatalf("Reading ISO from %v: got %v, want nil", v.isolink, err) because since I get an error here, the body will most likely be nil, thus, the next step would not work. Because md5.Sum(body which is nil) is not correct and valid. That should not work. Therefore, it would be Fatalf where it recognizes the error, provides it to the user, and then exits the user.